docs/docs/opinions.md

```diff
@@ -1,24 +1,59 @@
+> Don't _ever_ edit your raw data, especially not manually, and _especially_ not in Excel.
 
-Also, if data is immutable, it doesn't need source control in the same way that code does. Therefore, ***by default, the data folder is included in the `.gitignore` file.*** If you have a small amount of data that rarely changes, you may want to include the data in the repository. Github currently warns if files are over 50MB and rejects files over 100MB. Some other options for storing/syncing large data include [AWS S3](https://aws.amazon.com/s3/) with a syncing tool (e.g., [`s3cmd`](https://s3tools.org/s3cmd)), [Git Large File Storage](https://git-lfs.github.com/), [Git Annex](https://git-annex.branchable.com/), and [dat](https://dat-data.com/). Currently by default, we ask for an S3 bucket and use [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/s3/index.html) to sync data in the `data` folder with the server.
+The most important features of a quality data analysis are **correctness** and **reproducibility**—anyone should be able to re-run your analysis using only your module code and raw data and produce the same final products. The best way to ensure correctness is to test your analysis code. **The best way to ensure reproducibility is to treat your data analysis pipeline as a directed acyclic graph** ([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)), meaning that every result or conclusion in your analysis can be traced backwards to examine what combination of code and data it came from or run forwards to recreate any artifact of the analysis. 
 
-## Notebooks are for exploration and communication
+### Raw data is immutable
 
-Notebook packages like the [Jupyter notebook](https://jupyter.org/), [Zeppelin](https://zeppelin-project.org/), and other literate programming tools are very effective for exploratory data analysis. However, these tools can be less effective for reproducing an analysis. When we use notebooks in our work, we often subdivide the `notebooks` folder. For example, `notebooks/exploratory` contains initial explorations, whereas `notebooks/reports` is more polished work that can be exported as html to the `reports` directory.
+That proper data analysis is a DAG means that **raw data must be treated as immutable**—it's okay to read and copy raw data to manipulate it into new outputs, but never okay to change it in place. This informs the design of the default `data/` directory subfolders in which data originates from `raw` and `external`, intermediate analytical outputs get serialized or cached in `interim`, and final products end up in `processed` (the number or names of these folders is less important than flow of data between them). 
```

Comment:
```suggestion
That proper data analysis is a DAG means that **raw data must be treated as immutable**—it's okay to read and copy raw data to manipulate it into new outputs, but never okay to change it in place. This informs the design of the default `data/` directory subfolders in which data originates from `raw/` and `external/`, intermediate analytical outputs get serialized or cached in `interim/`, and final products end up in `processed/` (the number or names of these folders is less important than flow of data between them). 
```

We have trailing slashes inconsistently. Personally, I like them because they're a simple visual signpost that you're talking about a directory, but I can go either way on it.

Reply:
+1, I'll take a pass throughout to update